### ISSUE TITLE
Shaded Dependencies Causing Conflicts w/Guava

### ISSUE THREAD

**Issue by springframeworkguru** [spring-cloud/spring-cloud-contract#1041]
The shaded dependencies is bringing in a rather dated version of Guava, which is causing issues with libraries which depend on new versions of Guava. 

Inspecting the spring-cloud-contract-shade jar (2.1.1.RELEASE) shows Guava version 11.02 (circa Feb 2012) is packaged:
```
#Generated by Maven
#Wed Feb 22 09:01:50 EST 2012
version=11.0.2
groupId=com.google.guava
artifactId=guava
```
This dependency and several others are **not** being repacked. 

This is leading to other libraries picking up the older version and throwing the following exceptions when running contract tests. 
```
java.lang.NoSuchMethodError: com.google.common.base.CharMatcher.ascii()Lcom/google/common/base/CharMatcher;

	at com.google.common.net.MediaType.<clinit>(MediaType.java:85)
	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateMediaTypes(RequestValidator.java:129)
	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateContentType(RequestValidator.java:97)
	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateRequest(RequestValidator.java:85)
	at com.atlassian.oai.validator.OpenApiInteractionValidator.lambda$validate$0(OpenApiInteractionValidator.java:181)
	at java.util.function.Function.lambda$andThen$1(Function.java:88)
```
Or
```
java.lang.NoClassDefFoundError: Could not initialize class com.google.common.net.MediaType

	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateMediaTypes(RequestValidator.java:129)
	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateContentType(RequestValidator.java:97)
	at com.atlassian.oai.validator.interaction.request.RequestValidator.validateRequest(RequestValidator.java:85)
	at com.atlassian.oai.validator.OpenApiInteractionValidator.lambda$validate$0(OpenApiInteractionValidator.java:181)
```

Workaround is to exclude the SCC shaded jar:
```
          <exclusions>
                <exclusion>
                    <groupId>org.springframework.cloud</groupId>
                    <artifactId>spring-cloud-contract-shade</artifactId>
                </exclusion>
            </exclusions>
```

Ideal solution would be to not include transitive dependencies in the shaded jar.

**Comment by marcingrzejszczak** [comment]
This is interesting cause all of the dependencies are optional in the shaded JAR. Can you print the dependency tree please?

**Comment by marcingrzejszczak** [comment]
Hey @springframeworkguru can you check out the latest snapshots?

**Comment by mariuszs** [comment]
@marcingrzejszczak I have the same issue :(

```The following method did not exist:

    com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;CLjava/lang/Object;)V

The method's class, com.google.common.base.Preconditions, is available from the following locations:

    jar:file:/app/lib/spring-cloud-contract-shade-2.1.3.RELEASE.jar!/com/google/common/base/Preconditions.class
    jar:file:/app/lib/guava-28.1-jre.jar!/com/google/common/base/Preconditions.class

It was loaded from the following location:

    file:/app/lib/spring-cloud-contract-shade-2.1.3.RELEASE.jar


Action:

Correct the classpath of your application so that it contains a single, compatible version of com.google.common.base.Preconditions```

**Comment by marcingrzejszczak** [comment]
I think we fixed it in 2.2.0 but it seems I need to backport it to 2.1.x

**Comment by marcingrzejszczak** [comment]
Fixed via https://github.com/spring-cloud/spring-cloud-contract/commit/f6f51cdae661ff17e44babc662115d960024ce00
